core/src/test/java/org/apache/iceberg/TestDefaultValueParser.java

```diff
@@ -0,0 +1,145 @@
+
+  @Test
+  public void testValidDefaults() throws IOException {
+    Object[][] typesWithDefaults = new Object[][] {
+        {Types.BooleanType.get(), "null"},
+        {Types.BooleanType.get(), "true"},
+        {Types.IntegerType.get(), "1"},
+        {Types.LongType.get(), "9999999"},
+        {Types.FloatType.get(), "1.23"},
+        {Types.DoubleType.get(), "123.456"},
+        {Types.DateType.get(), "\"2007-12-03\""},
+        {Types.TimeType.get(), "\"10:15:30\""},
+        {Types.TimestampType.withoutZone(), "\"2007-12-03T10:15:30\""},
+        {Types.TimestampType.withZone(), "\"2007-12-03T10:15:30+01:00\""},
+        {Types.StringType.get(), "\"foo\""},
+        {Types.UUIDType.get(), "\"eb26bdb1-a1d8-4aa6-990e-da940875492c\""},
+        {Types.FixedType.ofLength(2), "\"111f\""},
+        {Types.BinaryType.get(), "\"0000ff\""},
+        {Types.DecimalType.of(9, 2), "123.45"},
+        {Types.ListType.ofOptional(1, Types.IntegerType.get()), "[1, 2, 3]"},
+        {Types.MapType.ofOptional(1, 2, Types.IntegerType.get(), Types.StringType.get()),
+         "{\"keys\": [1, 2], \"values\": [\"foo\", \"bar\"]}"},
+        {Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc")),
+         "{\"1\": 1, \"2\": \"bar\"}"},
+        // deeply nested complex types
+        {Types.ListType.ofOptional(1, Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc"))), "[{\"1\": 1, \"2\": \"bar\"}, {\"1\": 2, \"2\": " +
+             "\"foo\"}]"},
+        {Types.MapType.ofOptional(1, 2, Types.IntegerType.get(), Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc"))),
+         "{\"keys\": [1, 2], \"values\": [{\"1\": 1, \"2\": \"bar\"}, {\"1\": 2, \"2\": \"foo\"}]}"},
+        {Types.StructType.of(
+            required(1, "f1", Types.StructType.of(
+                optional(2, "ff1", Types.IntegerType.get(), "doc"),
+                optional(3, "ff2", Types.StringType.get(), "doc")), "doc"),
+            optional(4, "f2", Types.StructType.of(
+                optional(5, "ff1", Types.StringType.get(), "doc"),
+                optional(6, "ff2", Types.IntegerType.get(), "doc")), "doc")),
+         "{\"1\": {\"2\": 1, \"3\": \"bar\"}, \"4\": {\"5\": \"bar\", \"6\": 1}}"},
+        };
+
+    for (Object[] typeWithDefault : typesWithDefaults) {
+      Type type = (Type) typeWithDefault[0];
+      String defaultValue = (String) typeWithDefault[1];
+
+      String roundTripDefaultValue = defaultValueParseAndUnParseRoundTrip(type, defaultValue);
+      // Only if the type is a timestampWithZone type, the round-trip default value will always be standardized to the
+      // UTC time zone, which might be different in the original value, but they should represent the same instant
```

Comment:
I think this should probably reject values that have a non-zero offset, rather than allowing the serialized value to change.

Reply:
Sure, I adapted this change.

Reply:
@rdblue  There is one trivia worth mentioning, even the zero offset itself has 2 representations: `Z` or `+00:00` , and the java `DateTimeFormatter.ISO_OFFSET_DATE_TIME` will always output the `Z` representation. Do you mean to require users always set the timestamptz string to end with `Z` ?

Reply:
Thanks for asking! Sorry that I didn't see this, but I think we should use `+00:00` because that's the form in the spec.